src/debugger/include/debugger/cli_debugger.h

```diff
@@ -0,0 +1,30 @@
+/**
+ * @file cli_debugger.h
+ * Declares the CliDebugger Class for CLI Debugger Operations
+ */
+
+#pragma once
+
+#include "debugger.h"
+#include "debugger/debugger.fwd.h"
+#include "debugger/debugger_interface.h"
+
+namespace debugger {
+
+class CliDebugger : public IDebugger {
+  private:
+	std::shared_ptr<DebuggerCore> debugger_core;
```

Comment:
Does this need to be a `shared_ptr`? Since this is the sole owner of the ```DebuggerCore``` object, this can be made a ```unique_ptr```.